libs/application/template-api-modules/src/lib/modules/templates/driving-license-submission/driving-license-submission.service.ts

```diff
@@ -157,24 +159,37 @@ export class DrivingLicenseSubmissionService extends BaseTemplateApiService {
       .nationalId
     const teacherNationalId = application.applicant
 
-    const result = await this.drivingLicenseService.newDrivingAssessment(
-      studentNationalId as string,
-      teacherNationalId,
-    )
-
-    if (result.success) {
-      await this.sharedTemplateAPIService.sendEmail(
-        generateDrivingAssessmentApprovalEmail,
-        application,
-      )
-    } else {
-      throw new Error(
-        `Unexpected error (creating driver's license): '${result.errorMessage}'`,
+    try {
+      const result = await this.drivingLicenseService.newDrivingAssessment(
+        studentNationalId as string,
+        teacherNationalId,
       )
-    }
 
+      if (result.success) {
+        await this.sharedTemplateAPIService.sendEmail(
+          generateDrivingAssessmentApprovalEmail,
+          application,
+        )
+      } else {
+        throw new Error(
+          `Unexpected error (creating driver's license): '${result.errorMessage}'`,
+        )
+      }
+    } catch (e) {
+      if (e instanceof Error && e.name === 'FetchError') {
+        const err = (e as unknown) as FetchError
+        throw new TemplateApiError(
+          {
+            title:
+              err.problem?.title || coreErrorMessages.failedDataProviderSubmit,
+            summary: err.problem?.detail || '',
+          },
+          400,
+        )
+      }
+    }
     return {
-      success: result.success,
+      success: false,
```

Comment:
```javascript
return {
  success: false
}
```

at the function top level, after a try block wherein the try part returns nothing.
Does this not result in success always being reported as false or am I missing something?
Is this intentional perhaps?